### PR TITLE
Run pods needing control-plane instance credentials on hostNetwork

### PR DESCRIPTION
**Pull request by johngmyers**
Allows setting IMDS hop limit to 1 on control-plane nodes.

### CODE REVIEW

**Comment by johngmyers** [comment]
many-addons broken due to cluster-autoscaler

**Comment by rifelpet** [comment]
I believe this has a side effect of preventing CiliumNetworkPolicies from restricting the network access for these pods

**Comment by johngmyers** [comment]
@olemarkus What is with the upgrade-ab test? It's failing because the PDB is not satisfied 5 seconds after the new manifest was applied. It takes time for a Deployment to roll out.

We probably need a loop around the `${CHANNELS} apply channel` command, waiting for it to reconcile.

**Comment by johngmyers** [comment]
I don't see a problem with CiliumNetworkPolicies not applying to these pods. They are privileged, after all. If they care that much about restricting their access, they can enable IRSA.

File: pkg/wellknownports/wellknownports.go [hunk 1]
```diff
@@ -73,6 +73,21 @@ const (
...
+	AWSLBCHealthPort = 9441
+
+	//AWSLBCMetricsPort is reserved for the AWS Load Balancer Controller's metrics.
+	AWSLBCMetricsPort = 9442
+
+	// AWSLBCWebhookPort is reserved for the AWS Load Balancer Controller's webhook server.
+	AWSLBCWebhookPort = 9443
+
+	// CloudCSIDriverPort is reserved for the cloud provider's primary CSI driver.
+	CloudCSIDriverPort = 9808
```
Comment (hakman) [inline pkg/wellknownports/wellknownports.go:89]: This is only needed for AWS, no need to complicate other providers with it.
```suggestion
	// AWSCSIDriverPort is reserved for the cloud provider's primary CSI driver.
	AWSCSIDriverPort = 9808
```

Comment (johngmyers) [reply pkg/wellknownports/wellknownports.go:89]: The OpenStack cinder daemonset runs on host network and uses a port.

The GCP PD controller also runs on host network, though it uses three ports.

@justinsb should we not try to share port assignments across the primary CSI drivers for cloud providers?

Comment (johngmyers) [reply pkg/wellknownports/wellknownports.go:89]: We should be more proactive in avoiding port conflicts. We don't have the coverage to find them all through testing.

Comment (olemarkus) [reply pkg/wellknownports/wellknownports.go:89]: We probably should have a map of the ports in `wellknownports.go` and use that map in manifests that are using host networking. That would make it easier to discover any conflicts.

**Comment by johngmyers** [comment]
Making this PR minimal and leaving the avoidance of other port conflicts to future work.

**Review by hakman — APPROVED** [review]

**Pull request merged**